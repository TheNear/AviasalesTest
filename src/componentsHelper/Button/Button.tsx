import React, { ButtonHTMLAttributes } from "react";
import style from "./Button.module.css";

interface IButton extends ButtonHTMLAttributes<HTMLButtonElement> {
  text: string;
  active?: boolean;
}

const Button: React.FC<IButton> = ({ text, active }) => {
  return (
    <button type="button" className={`${style.button} ${active ? style.active : ""}`}>
      {text}
    </button>
  );
};

export default Button;
