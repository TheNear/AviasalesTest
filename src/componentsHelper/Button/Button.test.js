import React from "react";
import { shallow } from "enzyme";
import Button from "./Button";

describe("Common Button render", () => {
  it("Render empty button", () => {
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Render inactive success button", () => {
    const wrapper = shallow(<Button text="SUCCESS" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Render active reject button", () => {
    const wrapper = shallow(<Button text="REJECT" active />);
    expect(wrapper).toMatchSnapshot();
  });
});
