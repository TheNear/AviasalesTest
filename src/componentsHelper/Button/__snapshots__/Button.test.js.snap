// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common Button render Render active reject button 1`] = `
<button
  className="button active"
  type="button"
>
  REJECT
</button>
`;

exports[`Common Button render Render empty button 1`] = `
<button
  className="button "
  type="button"
/>
`;

exports[`Common Button render Render inactive success button 1`] = `
<button
  className="button "
  type="button"
>
  SUCCESS
</button>
`;
